sherpa/astro/data.py

```diff
@@ -1427,7 +1429,15 @@ def __init__(self, name, channel, counts, staterror=None, syserror=None,
         self.exposure = exposure
         self.backscal = backscal
         self.areascal = areascal
-        self.header = header
+        if header is None:
+            self.header = {'HDUCLASS': "OGIP", 'HDUCLAS1': "SPECTRUM",
```

Comment:
Some tests used to initialize the header like this. Note sure why... but instead it makes sense to do that here.